File: softpos-base-java-light/app/src/main/java/it/nexi/softposbase_java_light_01/activities/MainActivity.java
package it.nexi.softposbase_java_light_01.activities;

import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import it.nexi.softposbase_java_light_01.commons.Constant;
import it.nexi.softposbase_java_light_01.databinding.ActivityMainBinding;

/**
 *
 *  Nexi Payment
 *
 * The app's main activity.
 * It contains the menu that allows you
 * to launch the Nexi POS methods.
 *
 */
public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        try {

            super.onCreate(savedInstanceState);

            EdgeToEdge.enable(this);

            // Init ViewBinding
            binding = ActivityMainBinding.inflate(getLayoutInflater());

            setContentView(binding.getRoot());

            ViewCompat.setOnApplyWindowInsetsListener(binding.main, (v, insets) -> {
                Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
                v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
                return insets;
            });

            // Set app version
            String versionName ="Version : ";
            PackageInfo pInfo = null;
            try {
                pInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
                versionName += pInfo.versionName;
            } catch (PackageManager.NameNotFoundException e) {
                throw new RuntimeException(e);

            }

            binding.textViewVersioneApp.setText(versionName);

            binding.btnPayment.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {

                    Intent intentNew = new Intent(MainActivity.this, PaymentInsertValuesActivity.class);
                    startActivity(intentNew);
                }

            });

            binding.btnReversal.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {

                    Intent intentNew = new Intent(MainActivity.this, ReversalInsertValuesActivity.class);
                    startActivity(intentNew);

                }

            });

            binding.btnLastTransaction.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent intentNew = new Intent(MainActivity.this, LastTransactionInsertValuesActivity.class);
                    startActivity(intentNew);
                }

            });

            binding.btnCloseAccount.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent intentNew = new Intent(MainActivity.this, AccountingClosureInsertValuesActivity.class);
                    startActivity(intentNew);
                }

            });

        } catch (Exception ex){
            Log.e(Constant.LOG_PREF_MSG_ERR, ex.getMessage());
        }
    }


}

File: softpos-base-kotlin-demo-01/app/src/main/java/it/nexi/softposbase_kotlin_demo_01/activities/MainActivity.kt
package it.nexi.softposbase_kotlin_demo_01.activities

import android.content.Intent
import android.content.pm.PackageInfo
import android.os.Build
import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import it.nexi.softposbase_kotlin_demo_01.commons.Constant
import it.nexi.softposbase_kotlin_demo_01.databinding.ActivityMainBinding

/**
 *
 *  Nexi Payment
 *
 * The app's main activity.
 * It contains the menu that allows you
 * to launch the Nexi POS methods.
 *
 */
class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        try {

            super.onCreate(savedInstanceState)

            enableEdgeToEdge()

            binding = ActivityMainBinding.inflate(layoutInflater)

            setContentView(binding.root)

            ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v, insets ->
                val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
                v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
                insets
            }

            // Set the app version
            setAppVersion()

            // Configure click listener
            setupClickListeners()

        } catch (ex: Exception) {
            Log.e(Constant.LOG_PREFISSO_MSG_ERR, ex.message ?: "Unknown Error")
        }
    }

    private fun setAppVersion() {
        try {

            val pInfo: PackageInfo = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
                packageManager.getPackageInfo(packageName, android.content.pm.PackageManager.PackageInfoFlags.of(0))
            } else {
                @Suppress("DEPRECATION")
                packageManager.getPackageInfo(packageName, 0)
            }

            binding.textViewAppVersion.text = "Version: ${pInfo.versionName}"

        } catch (e: Exception) {
            Log.e(Constant.LOG_PREFISSO_MSG_ERR, "Error retrieving the version : ${e.message}")
        }
    }

    private fun setupClickListeners() {

        binding.apply {

            btnPayment.setOnClickListener {
                startActivity(
                    Intent(
                        this@MainActivity,
                        PaymentInsertValuesActivity::class.java
                    )
                )
            }

            btnReversal.setOnClickListener {
                startActivity(Intent(this@MainActivity, ReversalInsertValuesActivity::class.java))
            }

            btnLastTransaction.setOnClickListener {
                startActivity(Intent(this@MainActivity, LastTransactionInsertValuesActivity::class.java))
            }

            btnCloseAccount.setOnClickListener {
                startActivity(Intent(this@MainActivity, AccountingClosureInsertValuesActivity::class.java))
            }

        } // binding.apply {

    } //  private fun setupClickListeners() {

}
